ugs-platform/ugs-platform-visualizer/src/main/java/com/willwinder/ugs/nbm/visualizer/renderables/GcodeModel.java

```diff
@@ -158,13 +155,18 @@ public void draw(GLAutoDrawable drawable, boolean idle, Point3d machineCoord, Po
         GL2 gl = drawable.getGL().getGL2();
```

Comment:
This might be a bigger problem, I wonder if the whole renderable pipeline needs to be fixed somehow. Whatever initiates the loop to call draw on all the renderable a should be able to skip frames like you did in updateVertexBuffers